packages/js/src/settings/components/formik-page-select-field.js

```diff
@@ -0,0 +1,166 @@
+/* eslint-disable complexity */
+import { DocumentAddIcon } from "@heroicons/react/outline";
+import { useCallback, useEffect, useMemo, useState } from "@wordpress/element";
+import { __ } from "@wordpress/i18n";
+import { AutocompleteField, Spinner } from "@yoast/ui-library";
+import classNames from "classnames";
+import { useField } from "formik";
+import { debounce, find, isEmpty, map, values } from "lodash";
+import PropTypes from "prop-types";
+import { ASYNC_ACTION_STATUS } from "../constants";
+import { useDispatchSettings, useSelectSettings } from "../hooks";
+
+let abortController;
+
+
+/**
+ * @param {JSX.node} children The children.
+ * @param {string} [className] The className.
+ * @returns {JSX.Element} The page select options content decorator component.
+ */
+const PageSelectOptionsContent = ( { children, className = "" } ) => (
+	<div className={ classNames( "yst-flex yst-items-center yst-justify-center yst-gap-2 yst-py-2 yst-px-3", className ) }>
+		{ children }
+	</div>
+);
+
+PageSelectOptionsContent.propTypes = {
+	children: PropTypes.node.isRequired,
+	className: PropTypes.string,
+};
+
+/**
+ * @param {Object} props The props object.
+ * @param {string} props.name The field name.
+ * @param {string} props.id The field id.
+ * @param {string} props.className The className.
+ * @param {string} props.description The description.
+ * @returns {JSX.Element} The page select component.
+ */
+const FormikPageSelectField = ( { name, id, className = "", ...props } ) => {
+	const siteBasicsPolicies = useSelectSettings( "selectPreference", [], "siteBasicsPolicies", {} );
+	const pages = useSelectSettings( "selectPagesWith", [ siteBasicsPolicies ], values( siteBasicsPolicies ) );
+	const { addManyPages, fetchPages } = useDispatchSettings();
+	const [ { value, ...field }, , { setTouched, setValue } ] = useField( { type: "select", name, id, ...props } );
+	const [ status, setStatus ] = useState( ASYNC_ACTION_STATUS.idle );
+	const [ initialPagesFound, setInitialPagesFound ] = useState( false );
+	const [ queriedPageIds, setQueriedPageIds ] = useState( [] );
+	const canCreatePages = useSelectSettings( "selectPreference", [], "canCreatePages", false );
+	const createPageUrl = useSelectSettings( "selectPreference", [], "createPageUrl", "" );
+
+	const selectedPage = useMemo( () => {
+		const pageObjects = values( pages );
+		return find( pageObjects, [ "id", value ] );
+	}, [ value, pages ] );
+
+
+	const debouncedFetchPages = useCallback( debounce( async search => {
+		try {
+			setStatus( ASYNC_ACTION_STATUS.loading );
+			// Cleanup previous running request.
+			if ( abortController ) {
+				abortController?.abort();
+			}
+			abortController = new AbortController();
+			// eslint-disable-next-line camelcase
+			const response = await fetchPages( { search, per_page: 20 } );
```

Comment:
Ah the abortcontroller is not used. This relies on the WP native way of using the API. and that does not support abortcontrollers. The user component can use that because it is made to be only shown once on a page

Reply:
I think the lack of the AbortController is what causes the following issue:

When you type not quick enough, you have multiple requests running simultaneously:
![image](https://github.com/Yoast/wordpress-seo/assets/12400734/b350518b-7653-418b-bda2-4b948d606b43)

If you have a big enough site (or just slow) where the responses of those simultaneous requests are not instant, you'll have the annoying UX of new results overwriting the old results all of a sudden. In case that's not clear, you can watch the screengrab:
https://drive.google.com/file/d/126G-KHU7TYwummpBNloUpm53TBETCXeU/view?usp=sharing

As you can see for the users option, the same thing is being handled more gracefully.